### PR TITLE
refactor: added resize window signal to update while ssh

### LINKED ISSUE #2630: Unable to forward the window resize message to remote
When I use iterm2, and let `okteto` to connect to remote development container, I noticed that if I resize the iterm2 window, the remote is not receiving the resize message. The evidence is that the vim window is not rendering properly when I resize.

**To Reproduce**

1. Open iTerm2
2. `okteto up -f okteto.yml`
3. `vim` or any other full screen program
4. Resize iTerm2 and the `vim` output is messed up

**Expected behavior**
`vim` should re-render with proper output.

**Desktop (please complete the following information):**

- OS: macOS
- Version 12.3.1
- CLI 2.2.1

**Additional context**
I have messaged people in Kubernetes Slack, okteto channel, with a screen snapshot.

### PR DESCRIPTION
**Pull request by teresaromero**
Signed-off-by: Teresa Romero <teresa@okteto.com>

Fixes #2630 

## Proposed changes

- when running the ssh connection, add a signal channel to capture the resize or window event and update the session window with the new size
-

### CODE REVIEW

File: pkg/ssh/exec.go [hunk 1]
```diff
@@ -159,6 +161,23 @@ func Exec(ctx context.Context, iface string, remotePort int, tty bool, inR io.Re
 		}
 	}()
 
+	resize := make(chan os.Signal)
+	signal.Notify(resize, syscall.SIGWINCH)
```
Comment (jLopezbarb) [inline pkg/ssh/exec.go:165]: Needs to be compiled on windows. Check https://github.com/docker/cli/blob/c76cbdc6a1ba8fead63b5673728d3c1b02cb510b/cli/command/container/tty.go#L71

File: pkg/ssh/resize_window_windows.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+			oktetoLog.Infof("request for terminal size failed: %s", err)
+		}
+		for {
+			time.Sleep(time.Millisecond * 250)
+			ws, err := dockerterm.GetWinsize(os.Stdout.Fd())
+			if err != nil {
+				oktetoLog.Infof("request for terminal size failed: %s", err)
+			}
+			if prevWSize.Height != ws.Height || prevWSize.Width != ws.Width {
+				if err := dockerterm.SetWinsize(os.Stdout.Fd(), ws); err != nil {
```
Comment (jLopezbarb) [inline pkg/ssh/resize_window_windows.go:28]:
```suggestion
 			session.WindowChange(height, width)
```

File: pkg/ssh/resize_window_windows.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	"time"
+
+	dockerterm "github.com/moby/term"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"golang.org/x/crypto/ssh"
+)
+
+func resizeWindow(session *ssh.Session) {
+	go func() {
+		prevWSize, err := dockerterm.GetWinsize(os.Stdout.Fd())
```
Comment (jLopezbarb) [inline pkg/ssh/resize_window_windows.go:17]:
```suggestion
		width, height, err := term.GetSize(int(os.Stdout.Fd()))
```

File: pkg/ssh/resize_window_windows.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		if err != nil {
+			oktetoLog.Infof("request for terminal size failed: %s", err)
+		}
+		for {
+			time.Sleep(time.Millisecond * 250)
+			ws, err := dockerterm.GetWinsize(os.Stdout.Fd())
```
Comment (jLopezbarb) [inline pkg/ssh/resize_window_windows.go:23]:
```suggestion
		width, height, err := term.GetSize(int(os.Stdout.Fd()))
```

**Review by jLopezbarb — APPROVED** [review]

**Comment by AgustinRamiroDiaz** [comment]
@teresaromero Resising in windows is working correctly against an image with vim. I tried with my vscode terminal against a debian image where I installed vim. The refresh takes about half a second

**Review by jmacelroy — APPROVED** [review]

**Pull request merged**